lib/connect/ws.js

```diff
@@ -36,7 +36,7 @@ function buildBuilder (client, opts) {
 
   if (opts.protocol === 'wss') {
     wssProperties.forEach(function (prop) {
-      if (opts.hasOwnProperty(prop)) {
+      if (opts.hasOwnProperty(prop) && !opts.wsOptions.hasOwnProperty(prop)) {
```

Comment:
Oh okay, forget my previous comment. If this is really needed then `select-keys` might not be useful. Unless this step can be done first.